src/ua.c

```diff
@@ -786,6 +802,11 @@ int ua_accept(struct ua *ua, const struct sip_msg *msg)
 	if (!ua || !msg)
 		return EINVAL;
 
+	if (ua_find_call_msg(ua, msg)) {
+		warning("ua: call was already accepted\n");
+		return EINVAL;
+	}
+
```

Comment:
I would try to find a solution that does not involve a raw pointer.

You can add some logic in the application that looks up a call in **INCOMING** state.
This call is then passed to `ua_accept()` which will set the state to something else.
When the next module tries to call `ua_accept()` it will also lookup incoming calls,
and it will find none because the previous call was already accepted.

Reply:
Sorry, had to look again.

This is not possible, because the purpose of `ua_accept()` is to allocate the one and only call for the given `sip_msg`.